### PR TITLE
Added listDeleteMessage to produce readable message as to why deletin…

### PR DESCRIPTION
**Pull request by naphets123**
…g an entity is not possible

### CODE REVIEW

**Comment by naphets123** [comment]
https://trello.com/c/8QUCp2yB/391-fk-check-messages-are-not-readable-medium#

File: library/lib/list.php [hunk 1]
```diff
@@ -82,12 +82,24 @@ function listDelete($table, $ids, $uri = false, $fktables = null)
...
                             if (in_array($foreignkey['TABLE_NAME'], $fktables) && ('RESTRICT' == $foreignkey['DELETE_RULE'])) {
+                                if (db_fieldexists($foreignkey['TABLE_NAME'], 'label')) {
+                                    $namestring = ',b.label AS label';
+                                } elseif (db_fieldexists($foreignkey['TABLE_NAME'], 'naam')) {
+                                    $namestring = ',b.naam AS label';
+                                } elseif (db_fieldexists($foreignkey['TABLE_NAME'], 'name')) {
+                                    $namestring = ',b.name AS label';
+                                } elseif (db_fieldexists($foreignkey['TABLE_NAME'], 'firstname')) {
+                                    $namestring = ',concat(b.firstname," ",b.lastname) AS label';
+                                } elseif (2 == 2) {
```
Comment (HaGuesto) [inline library/lib/list.php]: What is this for?

File: library/lib/list.php [hunk 2]
```diff
@@ -82,12 +82,24 @@ function listDelete($table, $ids, $uri = false, $fktables = null)
...
+                                SELECT b.id as id'.$namestring.'
                                 FROM '.$table.' a, '.$foreignkey['TABLE_NAME'].' b 
                                 WHERE a.'.$foreignkey['REFERENCED_COLUMN_NAME'].' = b.'.$foreignkey['COLUMN_NAME'].' AND '.(db_fieldexists($foreignkey['TABLE_NAME'], 'deleted') ? '(NOT b.deleted OR b.deleted IS NULL) AND ' : '').' a.id = :id', ['id' => $id]);
                                 if (count($restricted)) {
-                                    return [false, 'The entry '.($restricted[0]['label'] ? $restricted[0]['label'] : $restricted[0]['id'].' of the table '.$foreignkey['TABLE_NAME']).' is still linked to this item.<br>Please edit or delete it first.', false];
+                                    //return [false, 'The entry '.($restricted[0]['label'] ? $restricted[0]['label'] : $restricted[0]['id'].' of the table '.$foreignkey['TABLE_NAME']).' is still linked to this item.<br>Please edit or delete it first.', false];
```
Comment (HaGuesto) [inline library/lib/list.php]: delete comment

File: library/lib/list.php [hunk 3]
```diff
@@ -82,12 +82,24 @@ function listDelete($table, $ids, $uri = false, $fktables = null)
                         foreach ($foreignkeys as $foreignkey) {
                             // Do we want to check the foreign key and does the foreign key restrict the delete?
                             if (in_array($foreignkey['TABLE_NAME'], $fktables) && ('RESTRICT' == $foreignkey['DELETE_RULE'])) {
+                                if (db_fieldexists($foreignkey['TABLE_NAME'], 'label')) {
```
Comment (HaGuesto) [inline library/lib/list.php]: nice and clear way to do this.

File: library/lib/list.php [hunk 4]
```diff
@@ -112,6 +124,32 @@ function listDelete($table, $ids, $uri = false, $fktables = null)
     }
 }
 
+function listDeleteMessage($table, $id, $foreignkey, $restricted)
+{
+    $table_name = [
+        'camps' => 'camp',
+        'locations' => 'wharehouse',
```
Comment (HaGuesto) [inline library/lib/list.php]: typo

File: library/lib/list.php [hunk 5]
```diff
@@ -112,6 +124,32 @@ function listDelete($table, $ids, $uri = false, $fktables = null)
...
+        'organisations' => 'organisation',
+    ];
+    $object_table_name = [
+        'people' => 'a beneficiary',
+        'products' => 'a product',
+        'library' => 'a library',
+        'locations' => 'a wharehouse',
```
Comment (HaGuesto) [inline library/lib/list.php]: typo

File: library/lib/list.php [hunk 6]
```diff
@@ -112,6 +124,32 @@ function listDelete($table, $ids, $uri = false, $fktables = null)
...
+        'cms_users' => ' an user',
+        'cms_usergroups' => 'an usergroup',
+    ];
+    $object_name = implode('', array_column($restricted, 'label'));
+    $id_name = ' with id '.implode('', array_column($restricted, 'id'));
+    if (!empty($object_name)) {
+        $object_name = ' called '.$object_name.' ';
+    }
+
+    return 'This '.$table_name[$table].' cannot be deleted/deactivated since '.$object_table_name[$foreignkey['TABLE_NAME']].''.$object_name.' '.$id_name.' is still active';
```
Comment (HaGuesto) [inline library/lib/list.php]: replace deleted/deactivated with removed. See trello card for this.

Comment (HaGuesto) [reply library/lib/list.php]: add at the end. 'Please edit or remove it first.'

Comment (HaGuesto) [reply library/lib/list.php]: I know that martin is using the message for his tests. Could you also run the cypress tests locally and update all references where this text is asserted?

File: library/lib/list.php (hunk 1 above)
Comment (naphets123) [reply library/lib/list.php]: Ok initially i did this with a switch statement, therefore the default case had to trigger here.

Comment (HaGuesto) [reply library/lib/list.php]: understood

File: library/lib/list.php (hunk 6 above)
Comment (HaGuesto) [reply library/lib/list.php]: it is in the user group tests. @DurkoMatko I will tag you where you have to change something.

**Comment by HaGuesto** [comment]
[FK check messages are not readable #medium](https://trello.com/c/8QUCp2yB/391-fk-check-messages-are-not-readable-medium)

File: library/lib/list.php (hunk 6 above)
Comment (DurkoMatko) [reply library/lib/list.php]: hmm...I think you've also fixed the browsertest @stephan :D It was enough to just add  'Please edit or remove it first.' to the noty text.

I've merged master to my branch and tested it and tests ran smoothly. That was the only issue @HaGuesto, right?

**Pull request merged**